README.md

```diff
@@ -28,61 +28,46 @@ OpenCensus libraries require Go 1.8 or later as it uses the convenience function
 
 ## Tags API
 
-### To create/retrieve a key
-A key is defined by its name. To use a key a user needs to know its name and type (currently only keys of type string are supported. Later keys of type int64 and bool will be supported). Calling CreateKeyString(...) multiple times with the same name returns the same key.
+Tags represent propagated key values. They can propagated context.Context in the same process or can be encoded
+to be transmitted on wire and decoded back to a TagSet at the destination.
 
-Create/retrieve key:
+### Getting a key by a name
+A key is defined by its name. To use a key a user needs to know its name and type. Currently, only keys of type string are supported.
+The library will support int64 and bool keys in the future.
 
-```go
-if key1, err := tags.CreateKeyString("keyNameID1"); err != nil {
-    // handle error
-}
-```
+See the [KeyStringByName][keystringbyname-ex] example.
 
-### To create a set of tags associated with keys
-TagSet is a set of tags. The following code demonstrates how to create a new TagSet from scratch:
+### Creating a set of tags associated with keys
+TagSet is a set of tags. Package tags provide a builder to create tag sets.
 
-```go
-tsb := NewTagSetBuilder(nil)
-tsb.InsertString(key1, "foo value")
-tsb.UpdateString(key1, "foo value2")
-tsb.UpsertString(key2, "bar value")
-tagsSet := tsb.Build()
-```
+See the [NewTagSetBuilder][newtagsetbuilder-ex] example.
 
-The methods on Builder can be chained.
-
-To create a new TagSet from an existing tag set oldTagSet:
+### Propagating a TagSet in a context
+To propagate a tag set to downstream methods and downstream RPCs, add a tag set
+to the current context. NewContext will return a copy of the current context,
+and put the tag set into the returned one.
+If there is already a tag set in the current context, it will be deleted and replaced.
```

Comment:
"deleted" for me means: If I still use the old Context somewhere else that means will no longer have access to the old TagSet?

I would just say:
it will be replaced (or overwritten).

Reply:
Done